### PR TITLE
Dependency View Extension

### PR DESCRIPTION
**Pull request by jasondaming**
Adds an easy to use GUI for installing, updating, and removing vendor dependencies.

### CODE REVIEW

**Review by sciencewhiz — COMMENTED** [review]
When you delete a vendordep, it should prompt you to rebuild like it does when you add one.

Any way to highlight vendordeps that are local vs online?

**Comment by ArchdukeTim** [comment]
Is it possible to add ability to quick-add vendor deps, such as the ones listed at https://docs.wpilib.org/en/stable/docs/software/vscode-overview/3rd-party-libraries.html?

**Comment by sciencewhiz** [comment]
> Is it possible to add ability to quick-add vendor deps

Can you explain what you mean? I think that's what this PR does

File: vscode-wpilib/eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// eslint.config.js
+module.exports =
+  {
+      ignores: [".media/"]
```
Comment (sciencewhiz) [inline vscode-wpilib/eslint.config.js:4]: Shouldn't have the `.` in front of media

Comment (jasondaming) [reply vscode-wpilib/eslint.config.js]: I am just going off this: https://eslint.org/docs/latest/use/configure/ignore#ignoring-directories

Comment (jasondaming) [reply vscode-wpilib/eslint.config.js]: looks like you were right per usual.  Thanks for the help!

Comment (sciencewhiz) [reply vscode-wpilib/eslint.config.js]: I think that example was for a hidden directory (in linux hidden directories are prefixed with a period)

File: vscode-wpilib/src/dependencyView.ts [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+  private viewInfo: IProjectInfo | undefined;
+  private disposables: vscode.Disposable[] = [];
+  private installedDeps: IJsonDependency[] = [];
+  private availableDeps: IJsonList[] = [];
+  private availableDepsList: IJsonList[] = [];
+  private onlineDeps: IJsonList[] = [];
+  private installedList: IDepInstalled[] = [];
+  private homeDeps: IJsonDependency[] = [];
+  private externalApi: IExternalAPI;
+  private ghURL = `https://raw.githubusercontent.com/jasondaming/vendor-json-repo/ctre2024/`;
```
Comment (sciencewhiz) [inline vscode-wpilib/src/dependencyView.ts:37]: This should change to the WPILIb repo before merging

**Review by sciencewhiz — CHANGES_REQUESTED** [review]

File: vscode-wpilib/media/wpilib.png
Comment (sciencewhiz) [inline vscode-wpilib/media/wpilib.png:1]: #683 adds the logo in the resources directory

File: vscode-wpilib/src/dependencyView.ts (hunk 2 above)
Comment (jasondaming) [reply vscode-wpilib/src/dependencyView.ts:37]: Should I do this now even though 2024 / 2025 don't exist currently?

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**